src/views/webview-app/use-connection-form.ts

```diff
@@ -1,65 +1,214 @@
-  const [isConnecting, setIsConnecting] = useState(false);
-  const [connectionFormOpened, setConnectionFormOpened] = useState(false);
-  const [connectionAttemptId, setConnectionAttemptId] = useState('');
-  const [connectionErrorMessage, setConnectionErrorMessage] = useState('');
+  const [
+    {
+      initialConnectionInfo,
+      isConnecting,
+      isConnectionFormOpen,
+      isEditingConnection,
+      connectionErrorMessage,
+    },
+    dispatch,
+  ] = useReducer(connectionFormReducer, {
+    ...getDefaultConnectionFormState(),
+  });
 
   useEffect(() => {
     const handleConnectResultResponse = (event) => {
       const message: MESSAGE_FROM_EXTENSION_TO_WEBVIEW = event.data;
       if (
         message.command === MESSAGE_TYPES.CONNECT_RESULT &&
-        message.connectionAttemptId === connectionAttemptId
+        message.connectionId === initialConnectionInfo.id
       ) {
-        setIsConnecting(false);
-        if (message.connectionSuccess) {
-          setConnectionFormOpened(false);
-        } else {
-          setConnectionErrorMessage(message.connectionMessage);
-        }
+        dispatch({
+          type: 'connection-result',
+          connectionSuccess: message.connectionSuccess,
+          connectionMessage: message.connectionMessage,
+        });
+      }
+    };
+    window.addEventListener('message', handleConnectResultResponse);
+    () => window.removeEventListener('message', handleConnectResultResponse);
+  }, [initialConnectionInfo]);
+
+  useEffect(() => {
+    const handleConnectResultResponse = (event) => {
+      const message: MESSAGE_FROM_EXTENSION_TO_WEBVIEW = event.data;
+      if (message.command === MESSAGE_TYPES.OPEN_EDIT_CONNECTION) {
+        dispatch({
+          type: 'open-edit-connection',
+          connectionInfo: {
+            id: message.connection.id,
+            favorite: {
+              name: message.connection.name,
+            },
+            connectionOptions: message.connection.connectionOptions,
+          },
+        });
       }
     };
     window.addEventListener('message', handleConnectResultResponse);
     () => window.removeEventListener('message', handleConnectResultResponse);
-  }, [connectionAttemptId]);
+  }, []);
 
   return {
-    connectionFormOpened,
+    isConnectionFormOpen,
     isConnecting,
+    initialConnectionInfo,
     connectionErrorMessage,
     openConnectionForm: () => {
-      setConnectionFormOpened(true);
+      dispatch({
+        type: 'open-connection-form',
+      });
       sendFormOpenedToExtension();
     },
     closeConnectionForm: () => {
-      setConnectionFormOpened(false);
-      setConnectionErrorMessage('');
+      dispatch({
+        type: 'close-connection-form',
+      });
     },
     handleCancelConnectClicked: () => {
       sendCancelConnectToExtension();
     },
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    handleSaveConnectionClicked: (connectionAttempt: {
```

Comment:
I'll make a connection-form pr and make this optional so we don't need this here anymore.